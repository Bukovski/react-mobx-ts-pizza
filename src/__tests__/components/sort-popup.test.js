import React from 'react';
import { act, render, screen } from "@testing-library/react";
import SortPopup from "../../components/sort-popup.component";
import { sortItems } from "../../pages/home.page";
import userEvent from "@testing-library/user-event";


describe("SortPopup Component", () => {
	it("should render, open sort popup & select one", () => {
		const mockClickSortType = jest.fn();
		
		const { container } = render(
			<SortPopup
				items={ sortItems }
				activeSortType={ sortItems[ 1 ].type }
				onClickSortType={ mockClickSortType }
			/>
		);
		
		expect(mockClickSortType).toBeCalledTimes(0);
		expect(screen.queryByTestId("sort-popup")).not.toBeInTheDocument();
		
		expect(screen.getByText("цене").closest('div')).toHaveClass('sort__label');
		
		userEvent.click(screen.getByTestId("sort-visible"));
		expect(screen.queryByTestId("sort-popup")).toBeInTheDocument();
		
		expect(container.firstChild).toMatchSnapshot()
		
		userEvent.click(screen.getByText("популярности"));
		expect(screen.queryByTestId("sort-popup")).not.toBeInTheDocument();
		expect(mockClickSortType).toBeCalledTimes(1);
		
		userEvent.click(screen.getByTestId("sort-visible"));
		expect(screen.queryByTestId("sort-popup")).toBeInTheDocument();
		
		act(() => {
			userEvent.click(document.body);
		});
		expect(screen.queryByTestId("sort-popup")).not.toBeInTheDocument();
	})
	
	it("render without onClickSortType open sort popup & select one", () => {
		render(
			<SortPopup
				items={ sortItems }
				activeSortType={ sortItems[ 1 ].type }
			/>
		);
		
		expect(screen.queryByTestId("sort-popup")).not.toBeInTheDocument();
		
		userEvent.click(screen.getByTestId("sort-visible"));
		expect(screen.queryByTestId("sort-popup")).toBeInTheDocument();
		
		userEvent.click(screen.getByText("популярности"));
		expect(screen.queryByTestId("sort-popup")).not.toBeInTheDocument();
	})
});
