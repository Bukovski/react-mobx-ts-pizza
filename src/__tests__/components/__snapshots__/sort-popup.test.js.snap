// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortPopup Component should render, open sort popup & select one 1`] = `
<div
  class="sort"
>
  <div
    class="sort__label"
  >
    <svg
      class="rotated"
      fill="none"
      height="6"
      viewBox="0 0 10 6"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 5C10 5.16927 9.93815 5.31576 9.81445 5.43945C9.69075 5.56315 9.54427 5.625 9.375 5.625H0.625C0.455729 5.625 0.309245 5.56315 0.185547 5.43945C0.061849 5.31576 0 5.16927 0 5C0 4.83073 0.061849 4.68424 0.185547 4.56055L4.56055 0.185547C4.68424 0.061849 4.83073 0 5 0C5.16927 0 5.31576 0.061849 5.43945 0.185547L9.81445 4.56055C9.93815 4.68424 10 4.83073 10 5Z"
        fill="#2C2C2C"
      />
    </svg>
    <b>
      Сортировка по:
    </b>
    <span
      data-testid="sort-visible"
    >
      цене
    </span>
  </div>
  <div
    class="sort__popup"
    data-testid="sort-popup"
  >
    <ul>
      <li
        class=""
        data-testid="sort-popup-item"
      >
        популярности
      </li>
      <li
        class="active"
        data-testid="sort-popup-item"
      >
        цене
      </li>
      <li
        class=""
        data-testid="sort-popup-item"
      >
        алфавит
      </li>
    </ul>
  </div>
</div>
`;
